### DISCUSSION TITLE
Can Mikro ORM Ignore Undefined Fields on Update Operations?

### DISCUSSION THREAD

**Discussion by tufailra97** [mikro-orm/mikro-orm#4709] [Q&A] [answered]
I'm using mikro orm for a project and have a question regarding its behavior during update operations. In my application, I have a CustomerEntity schema defined as follows:

```ts
@Entity({ tableName: 'customers' })
export class CustomerEntity extends BaseEntity {
  @Property()
  firstName: string;

  @Property()
  lastName: string;

  @Property({ unique: true })
  email: string;

  @Property()
  password: string;}
```

In my usage example, when updating the entity, I sometimes have fields in my update DTOs that are undefined. For instance:

```ts
const user = await this.em.findOne(CustomerEntity, id);

user.firstName = updateUserDto.firstName;
user.lastName = updateUserDto.lastName; // This field can be undefined
```
Does Mikro ORM have a built-in feature or configuration option that allows it to ignore undefined fields during update operations and prevent them from being set to null automatically? In the example above, I want Mikro ORM to update only the fields that have non-undefined values in the updateUserDto object and not automatically set them to null.

**Comment by B4nan** [comment, marked as answer]
You could do this:

```ts
const user = await this.em.findOne(CustomerEntity, id);
// updateUserDto = { ...updateUserDto }; // maybe not needed, depends on the context
Utils.dropUndefinedProperties(updateUserDto);
thks.em.assign(user, updateUserDto);
```

We could have an option in the `assign` method to do this under the hood.

**Reply by hayksaryan** [reply]
I think this should also be globally configurable.

**Reply by B4nan** [reply]
`assign` can be used as a static method, as such it cannot respect any global configs

and if you mean for what the OP is asking, I'd argue its very weird pattern to assign things, and expect that assigning `undefined` will have no meaning. if there was a value, assigning `undefined` means you changed the value. but we can surely have an option in the `assign` to do this automatically, as i suggested

**Reply by ruscon** [reply]
@B4nan 
Have the same issue.

> We could have an option in the assign method to do this under the hood.

That would be nice. 

> I'd argue its very weird pattern to assign things, and expect that assigning undefined will have no meaning. if there was a value, assigning undefined means you changed the value. but we can surely have an option in the assign to do this automatically, as i suggested

I use the `class-transformer` and after calling `plainToInstance` the optional properties turn out to be `undefined`.
Moreover, in order to remove `undefined` values, need to call `instanceToPlain` with specific `exposeUnsetFields: false` options.

As I understand it, in general when working with `mikro-orm`, it doesn't care whether I use POJOs or not, except in the case of `AssignOptions.mergeObjects` option
